core/confsys/service.go

```diff
@@ -129,19 +125,8 @@ func (s *Service) NewRunNumber() (runNumber uint32, err error) {
 	} else {
 		// Unsafe check-and-set, only for file backend
 		var rnf string
-		rnf, err = s.src.Get("o2/control/run_number_file")
-		if err != nil {
-			return
-		}
-		rnf, err = homedir.Expand(rnf) // Resolve ~ into home dir
-		if err != nil {
-			return
-		}
+		rnf = viper.GetString("coreWorkingDir") + "runcounter.txt"
```

Comment:
Are we certain that `coreWorkingDir` has a trailing slash?
Generally it shouldn't: the default shouldn't have a trailing `/`, and if the user passes a `/path/with/trailing/slash/`, it should be sanitized to remove it.

Then, this line becomes
```
		rnf = viper.GetString("coreWorkingDir") + "/runcounter.txt"
```

Reply:
Addendum: all user-provided paths must be sanitized to ensure the absence of a trailing slash.

Reply:
The `coreWorkingDir` is sanitized [here](https://github.com/AliceO2Group/Control/blob/master/core/config.go#L184).  I followed the same pattern used for the `repositoriesPath`.